src/main/reactjs/src/components/contactRequest/ContactRequestForm.tsx

```diff
@@ -0,0 +1,171 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+
+import { H1, Text } from 'components/elements/Text';
+import { useAppTranslation } from 'configs/i18n';
+import { useAppDispatch, useAppSelector } from 'configs/redux';
+import {
+  sendContactRequest,
+  setContactRequest,
+} from 'redux/actions/contactRequest';
+import { ContactRequest } from 'interfaces/contactRequest';
+import { APIResponseStatus } from 'enums/api';
+import { ProgressIndicator } from 'components/elements/ProgressIndicator';
+import { publicTranslatorsSelector } from 'redux/selectors/translatorDetails';
+import { contactRequestSelector } from 'redux/selectors/contactRequest';
+import {
+  CancelRequestDialog,
+  ErrorDialogWrapper,
+  SuccessDialogWrapper,
+} from 'components/contactRequest/ConnectedDialog';
+import {
+  VerifyTranslatorsStep,
+  FillContactDetailsStep,
+  WriteMessageStep,
+  PreviewAndSendStep,
+  ContactRequestStepper,
+} from 'components/contactRequest/Steps';
+
+const decrementStep = (step: number) => step - 1;
+const incrementStep = (step: number) => step + 1;
+
+const ControlButtons = ({
+  onCancelRequest,
+  onChangeStep,
+  step,
+  maxStep,
+  disableNext,
+}: {
+  onCancelRequest: () => void;
+  onChangeStep: Dispatch<SetStateAction<number>>;
+  step: number;
+  maxStep: number;
+  disableNext: boolean;
+}) => {
+  const { t } = useAppTranslation({
+    keyPrefix: 'akt.component.contactRequestForm',
+  });
+
+  const dispatch = useAppDispatch();
+  const request = useAppSelector(contactRequestSelector)
+    .request as ContactRequest;
+  const submit = () => {
+    dispatch(sendContactRequest(request));
+  };
+  return (
+    <div className="columns flex-end gapped">
+      <Button variant="outlined" color="secondary" onClick={onCancelRequest}>
+        {t('buttons.cancel')}
+      </Button>
+      <Button
+        variant="contained"
+        color="secondary"
+        onClick={() => onChangeStep(decrementStep)}
+        disabled={step == 0}
+      >
+        {t('buttons.previous')}
+      </Button>
+      {step == maxStep ? (
+        <Button variant="contained" color="secondary" onClick={() => submit()}>
+          {t('buttons.submit')}
+        </Button>
+      ) : (
+        <Button
+          variant="contained"
+          color="secondary"
+          disabled={disableNext}
+          onClick={() => onChangeStep(incrementStep)}
+        >
+          {t('buttons.next')}
+        </Button>
+      )}
+    </div>
+  );
+};
+
+const useResetContactRequestState = () => {
+  const dispatch = useAppDispatch();
+  const { filters, selectedTranslators } = useAppSelector(
+    publicTranslatorsSelector
+  );
+  const { fromLang: from, toLang: to } = filters;
+  useEffect(() => {
+    dispatch(
+      setContactRequest({
+        languagePair: { from, to },
+        translatorIds: selectedTranslators,
+        email: '',
+        firstName: '',
+        lastName: '',
+        message: '',
+        phoneNumber: '',
```

Comment:
Voisiko asettaa näitä contactRequest reducerissa alussa ns. defaultStatena? Silloin näitä ei tarvitse asettaa enää tässä.

Reply:
Joo, varmaankin. Ehkä pitää tietomallia muokata jotenkin, koska tosiaan mm. tuo kielipari on nyt ContactRequestin pakollinen kenttä, ja sitähän ei tiedetä ennen kuin käyttäjä on tehnyt valinnan. :) Muut kentät voi varmaan olla tyhjää stringiä tai tyhjää arrayta defaultina.